docs/zkapps/o1js/common-types-and-functions.md

```diff
@@ -0,0 +1,354 @@
+
+// Assertions
+
+x.assertEquals(45); // True - passes
+x.assertGreaterThan(45); // False - throws
+x.assertGreaterThanOrEquals(45); // True - passes
+x.assertLessThan(45); // False - throws
+x.assertLessThanOrEquals(45); // True - passes
+
+// Converting to Field
+
+const y = x.value; // This is a Field with the value 45
+```
+
+## Public and Private Keys
+
+For accessing Mina blockchain accounts, you can use `PublicKey` and `PrivateKey` types of o1js. You can create a key from a [base58 string](https://en.wikipedia.org/wiki/Binary-to-text_encoding).
+
+<!-- prettier-ignore -->
+```ts
+const x_private = PrivateKey.fromBase58("base58_private_key");
+const y_private = PrivateKey.fromJSON("base58_private_key");
+
+const x = PublicKey.fromBase58("B62qq8sm7JdsED6VuDKNWKLAi1Tvz1jrnffuud5gXMq3mgtd");
+const y = PublicKey.fromJSON("B62qq8sm7JdsED6VuDKNWKLAi1Tvz1jrnffuud5gXMq3mgtd");
+```
+
+You can also create a random `PrivateKey`, an empty `PublicKey`, or convert a `PrivateKey` to a `PublicKey`.
+
+<!-- prettier-ignore -->
+```ts
+const random = PrivateKey.random();
+const empty = PublicKey.empty();
+
+const fromPrivKey = PublicKey.fromPrivateKey(random); // or equivalently random.toPublicKey()
+```
+
+# Conditionals
+
+Traditional conditional statements are not supported by o1js. This is due to the fact that o1js creates ZKPs, and conditional evaluation does not make sense in a ZKP. ZKPs are evaluated into mathematical statements when compiled by o1js, thus you cannot chose to evaluate a statement based on a truth value unknown at the compile time.
+
+<!-- prettier-ignore -->
+```ts
+// This will NOT work
+if (foo) {
+  x.assertEquals(y);
+}
+```
+
+Nevertheless, you can use the o1js built-in `Provable.if()` method, which is a ternary operator:
+
+<!-- prettier-ignore -->
+```ts
+const x = Provable.if(
+  Bool(foo), // The condition to evaluate inside if, must be of type Bool
+  a, // You can return any provable type from Provable.if
+  b // a and b must be of the same provable type
+); // Behaves like `foo ? a : b`
+```
+
+`Provable.if()` allows you to include any logical computation you need inside your ZKP. There is nothing wrong with evaluating more complex expressions inside `Provable.if()` and returning the final result.
+
+However, it works differently from ternary statements in the other languages and frameworks. Both arguments of `Provable.if()` _always_ executes, regardless of the truth value of the condition. `Provable.if()` choses which one to return based on the truth value and sets the new variable. This is why you should never assign anything outside of the if scope or assert anything.
+
+<!-- prettier-ignore -->
+```ts
+const x = Provable.if(
+  Bool(foo),
+  (() => { // Here, we use arrow functions to return a Field element, but you can use any function that returns a provable type
+    return Field(5).div(3);
+  })(),
+  (() => {
+    return Field(5).div(2);
+  })()
+); // This variable is of type Field
```

Comment:
I dislike this way of presenting provable.if, because it for newbs it creates the impression that code is executed conditionally (I've experienced this)

Also, it's not cleaner than computing the two outcomes before selecting them with if. I recommend to show an example for that instead